Add documentation skipRecalibration and maintenance group (#404)
icub-tech-iit/documentation: docs/icub_robot_configuration/calibrator_configuration/calibrator_configuration.md

Relatively to the calibration procedure, users might configure an optional group in the `general.xml` file. Specifically the group has name `MAINTENANCE` and it currently holds just the `skipRecalibration` parameter.
In order to enable it, you should add the following group in general:

```xml
<group name="MAINTENANCE">
    <param extern-name="skipRecalibration" name="skipRecalibration"> false </param>
</group>
```

As you can observe, the parameter can be defined with `extern-name`. This means that you can overwrite its value when running the `yarprobotinterface` command as:

```sh
  yarprobotinterface --skipRecalibration true
```

Regarding the specifics of this parameter it can be described as follows:

- `skipRecalibration`, it can be set to `true` or `false` (default value). If set to `true` the robot will **skip** the calibration procedure when the `yarprobotinterface` is restarted while the boards are not switched off. Regarding this feature it is necessary to add some notes:
    1. The first time that the control board devices are started through the `yarprobotinterface` the calibration procedure runs normally, independently of having set the parameter to `true` or `false`

>     2. For all of the subsequent times if the `yarprobotinterface` gets restarted keeping the boards powered on the whole robot will skip the calibration procedure on all joints and they will be set to `IDLE`.

Reviewer: ```suggestion
    2. Subsequently, if `yarprobotinterface` gets restarted while keeping the boards powered on, then the whole robot will skip the calibration procedure on all joints, which will be set to `IDLE`.
```

>     3. Since the group and the parameter are not mandatory for running the robot one can omit it from the `general.xml` file and the software will set it to `false` by default, therefore the robot subparts will calibrate each time the devices are restarted as it normally happens.

Reviewer: ```suggestion
    3. Since the group and the parameter are not mandatory for running the robot, one can omit defining the group inside the `general.xml` file and the software will set it to `false` by default; therefore, the robot subparts will calibrate each time the devices are restarted, as it normally happens.
```

>     4. If some joints won't calibrate at the first start of the robot they won't skip the calibration procedure on the following runs for just those joints. Thus they will normally calibrate while all the others will be set to `IDLE`

Reviewer: ```suggestion
    4. If some joints won't calibrate at the first start of the robot, they won't skip the calibration procedure on the following runs. Thus, those joints will normally calibrate while all the others will be set to `IDLE`.
```

    5. If the parameter is set to `true` but the boards are restarted, the first time the robot devices are run the calibration procedure will execute normally.
    6. As defined above, since the parameter is configured using an `extern-name`, its value can be overwritten when running the `yarprobotinterface` without modifying the configuration files.

!!! warning

>     `skipRecalibration` is a feature thought to be used by expert users and it should be used with caution. Specifically, as detailed at `point d`, in the unlikely case that a joint won't calibrate during the first `yarprobotinterface` run, if this feature is enabled, it will calibrate during the subsequent run. Thus, since all the other joints will be set to `IDLE`, it might happen that the calibrating joint will collide with the others.

Author: oks, thanks.
in that case, I've written point d and not the number since when the documentation is generated online, it shows the numeration by letters and not numbers. That is the reason.

> !!! warning
>     `skipRecalibration` is a feature thought to be used by expert users and it should be used with caution. Specifically, as detailed at `point d`, in the unlikely case that a joint won't calibrate during the first `yarprobotinterface` run, if this feature is enabled, it will calibrate during the subsequent run. Thus, since all the other joints will be set to `IDLE`, it might happen that the calibrating joint will collide with the others.

Reviewer: ```suggestion
!!! danger

    `skipRecalibration` is a feature thought to be used by expert users, and it must be used with caution. In particular, in the unlikely event that a joint won't calibrate during the first `yarprobotinterface` run, if this feature is enabled, the same joint will keep calibrating during the subsequent runs. Thus, collisions may occur among joints in idle and the joint calibrating.
```

>     Therefore, when this feature is enabled, it is assumed that one is fully aware of how the joints move during the calibration, the calibration order of the subparts and which might be possible collisions if the case described at `point d` takes place.

Author: As above for the numeration. The other suggestions, instead, look like a good improvement to readability

>     Therefore, when this feature is enabled, it is assumed that one is fully aware of how the joints move during the calibration, the calibration order of the subparts and which might be possible collisions if the case described at `point d` takes place.

Reviewer: ```suggestion
```

>     Moreover, since at the restart all joints are set to `IDLE`, one should keep this in mind if the robot is started with a certain configuration that necessitate to have the joints controlled. 
>     Then, after the robot starts in `IDLE`, one can move all the joint in `POSITION` or other control mode and use it normally.

Reviewer: ```suggestion
```
